### PR TITLE
Implement Basic/misc.swift on Windows

### PR DESCRIPTION
**Pull request by jmittert**

### CODE REVIEW

File: Sources/Basic/misc.swift [hunk 1]
```diff
@@ -18,9 +18,20 @@ import Foundation
 ///   - args: The executable arguments.
 public func exec(path: String, args: [String]) throws {
     let cArgs = CStringArray(args)
+    #if os(Windows)
```
Comment (aciidb0mb3r) [inline Sources/Basic/misc.swift]:
```suggestion
  #if os(Windows)
```

File: Sources/Basic/misc.swift [hunk 2]
```diff
@@ -18,9 +18,20 @@ import Foundation
...
+    guard cArgs.cArray.withUnsafeBufferPointer({
+        $0.withMemoryRebound(to: UnsafePointer<Int8>?.self, {
+          _execv(path, $0.baseAddress) != -1
+        })
+    })
+    else {
+        throw SystemError.exec(errno, path: path, args: args)
+    }
+    #else
```
Comment (aciidb0mb3r) [inline Sources/Basic/misc.swift]:
```suggestion
  #else
```

File: Sources/Basic/misc.swift [hunk 3]
```diff
@@ -18,9 +18,20 @@ import Foundation
...
     guard execv(path, cArgs.cArray) != -1 else {
         throw SystemError.exec(errno, path: path, args: args)
     }
+    #endif
```
Comment (aciidb0mb3r) [inline Sources/Basic/misc.swift]:
```suggestion
  #endif
```

File: Sources/Basic/misc.swift [hunk 4]
```diff
@@ -163,13 +174,23 @@ public enum SystemError: Swift.Error {
...
+#else
 import func SPMLibc.strerror_r
+#endif
 import var SPMLibc.EINVAL
 import var SPMLibc.ERANGE
 
 extension SystemError: CustomStringConvertible {
     public var description: String {
         func strerror(_ errno: Int32) -> String {
+#if os(Windows)
```
Comment (aciidb0mb3r) [inline Sources/Basic/misc.swift]:
```suggestion
          #if os(Windows)
```

File: Sources/Basic/misc.swift [hunk 5]
```diff
@@ -163,13 +174,23 @@ public enum SystemError: Swift.Error {
...
+            let cap = 128
+            var buf = [Int8](repeating: 0, count: cap)
+            let _ = SPMLibc.strerror_s(&buf, 128, errno)
+            return "\(String(cString: buf)) (\(errno))"
+#else
```
Comment (aciidb0mb3r) [inline Sources/Basic/misc.swift]:
```suggestion
          #else
```

File: Sources/Basic/misc.swift [hunk 6]
```diff
@@ -187,6 +208,7 @@ extension SystemError: CustomStringConvertible {
                 return "\(String(cString: buf)) (\(errno))"
             }
             fatalError("strerror_r error: \(ERANGE)")
+#endif
```
Comment (aciidb0mb3r) [inline Sources/Basic/misc.swift]:
```suggestion
          #endif
```

**Comment by jmittert** [comment]
Fixed indents

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test

**Pull request merged**